File: front/bookland/src/app/models/UserAchievement.ts
import { User } from "./user";

export interface UserAchievement {
  id?: number;
  earned_at?: Date;
  achievement_id?: number;
   name: string;
  description: string;
  
  user_id?: number;
  iconUrl: string; 

  achievement?: {
  id?: number;
  name: string;
  description: string;
  tier: string;  // Make sure this exists
  iconUrl: string;
  };
}

File: Backend/PFE/src/main/java/com/example/PFE/user/UserAchievement.java
package com.example.PFE.user;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
public class UserAchievement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "achievement_id")
    private Achievement achievement;

    private LocalDateTime earnedAt;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Achievement getAchievement() {
        return achievement;
    }

    public void setAchievement(Achievement achievement) {
        this.achievement = achievement;
    }

    public LocalDateTime getEarnedAt() {
        return earnedAt;
    }

    public void setEarnedAt(LocalDateTime earnedAt) {
        this.earnedAt = earnedAt;
    }
}
